src/Campaigns/Blocks/CampaignGoal/edit.tsx

```diff
@@ -0,0 +1,105 @@
+import {__} from '@wordpress/i18n';
+import {useSelect} from '@wordpress/data';
+import {InspectorControls, useBlockProps} from '@wordpress/block-editor';
+import {BlockEditProps} from '@wordpress/blocks';
+import {ExternalLink, PanelBody, TextControl} from '@wordpress/components';
+import useCampaign from '../shared/hooks/useCampaign';
+import {CampaignSelector} from '../shared/components/CampaignSelector';
+
+import './styles.scss';
+
+const getGoalDescription = (goalType: string) => {
+    switch (goalType) {
+        case 'amount':
+            return __('Amount raised', 'give');
+        case 'donations':
+            return __('Number of donations', 'give');
+        case 'donors':
+            return __('Number of donors', 'give');
+        case 'amountFromSubscriptions':
+            return __('Recurring amount raised', 'give');
+        case 'subscriptions':
+            return __('Number of recurring donations', 'give');
+        case 'donorsFromSubscriptions':
+            return __('Number of recurring donors', 'give');
+    }
+}
+
+const getValue = (goalType: string, value: number) => {
+    switch (goalType) {
+        case 'amount':
+        case 'amountFromSubscriptions':
+            return currency.format(value);
+        default:
+            return value;
+    }
+}
+
+const currency = new Intl.NumberFormat('en-US', {
```

Comment:
Could we get the language string from `navigator`?

Reply:
Resolved https://github.com/impress-org/givewp/pull/7702/commits/e2cd89482b26b90150eb78098d2092ac0610bfb2